### ISSUE TITLE
R 4.0.0 support by rpy2 3.3.2 (?)

### ISSUE THREAD

**Issue by HanOostdijk** [rpy2/rpy2#695]
I ran into a problem when I try to use R from a Jupyter notebook (in Windows 10).
But let's start to say that I have no experience with Jupyter and very little with Python. 
Given my limited knowledge of these products I wonder if I made installation errors or that possibly my software versions are not yet supported
I installed (the most?) recent versions of Jupyter (1.0) , Python (3.8.2) , rpy2 (3.3.2) and R (4.0.0).
_Also with an earlier version of R I could not get rpy2 working without error messages (see end of this issue)._
I added `D:/tools/R/R-4.0.0/library/grDevices/libs/x64/` to the path when I saw the error messages but without result.

I include the jupyter and console output. If anyone can help me out: TIA
 
**Jupyter input and output:** 
![image](https://user-images.githubusercontent.com/13506389/80501947-58f4c480-8970-11ea-93ca-580573de516c.png)
![image](https://user-images.githubusercontent.com/13506389/80502190-9bb69c80-8970-11ea-87da-075b351ec1b3.png)
![image](https://user-images.githubusercontent.com/13506389/80502459-f3ed9e80-8970-11ea-9003-9399dfcf8539.png)
**Messages in command window:**
```
[I 16:11:48.235 NotebookApp] Kernel started: 571c2282-4870-4ff0-a05f-59b01937d92c
Warning message:
package 'methods' was built under R version 4.0.0
Error: package or namespace load failed for 'grDevices' in inDL(x, as.logical(local), as.logical(now), ...):
 unable to load shared object 'D:/tools/R/R-4.0.0/library/grDevices/libs/x64/grDevices.dll':
  LoadLibrary failure:  The specified procedure could not be found.

Error: package or namespace load failed for 'graphics' in inDL(x, as.logical(local), as.logical(now), ...):
 unable to load shared object 'D:/tools/R/R-4.0.0/library/grDevices/libs/x64/grDevices.dll':
  LoadLibrary failure:  The specified procedure could not be found.

Error: package or namespace load failed for 'stats' in inDL(x, as.logical(local), as.logical(now), ...):
 unable to load shared object 'D:/tools/R/R-4.0.0/library/grDevices/libs/x64/grDevices.dll':
  LoadLibrary failure:  The specified procedure could not be found.

During startup - Warning messages:
1: package 'datasets' was built under R version 4.0.0
2: package 'utils' was built under R version 4.0.0
3: package 'grDevices' was built under R version 4.0.0
4: package 'grDevices' in options("defaultPackages") was not found
5: package 'graphics' was built under R version 4.0.0
6: package 'graphics' in options("defaultPackages") was not found
7: package 'stats' was built under R version 4.0.0
8: package 'stats' in options("defaultPackages") was not found
Error: package or namespace load failed for 'grDevices' in inDL(x, as.logical(local), as.logical(now), ...):
 unable to load shared object 'D:/tools/R/R-4.0.0/library/grDevices/libs/x64/grDevices.dll':
  LoadLibrary failure:  The specified procedure could not be found.

Error in inDL(x, as.logical(local), as.logical(now), ...) :
  unable to load shared object 'D:/tools/R/R-4.0.0/library/grDevices/libs/x64/grDevices.dll':
  LoadLibrary failure:  The specified procedure could not be found.
In addition: Warning message:
package 'tools' was built under R version 4.0.0
[I 16:13:48.138 NotebookApp] Saving file at /hannotebook.ipynb
```
**Error message(s) in pop-up window:**
```
Python.exe Entry Point Not Found
The procedure entry point Rf_translateCharFP could not be found in the dynamic link library  D:\tools\R\R-4.0.0\library\grDevices\libs\x64\grDevices.dll
```
**Earlier R version:**
Not to confuse the issue but also with an earlier version of R I could not get it correctly working:
with R 3.6.0 and the same other versions (Jupyter (1.0) , Python (3.8.2) , rpy2 (3.3.2) ) I got a 'UserWarning'  in
```
D:\tools\python\python38\Lib\site-packages\rpy2\robjects\packages.py : 365
UserWarning: The symbol 'quartz' is not in this R namespace/package
    warnings.warn(
```

**Comment by HanOostdijk** [comment]
I see that R version 4.0.0. is still a work in progress: [Add R 4.0 to build matrix #697](https://github.com/rpy2/rpy2/pull/697). 
Thanks for all the work!

In the documentation about this change in [Python package / build (3.8, 4, windows-latest)](https://github.com/rpy2/rpy2/runs/635339153?check_suite_focus=true) I saw in the `install package` section some warnings/errors that made me wonder if the build was okay (even when the documentation calls the build successful) :

> 13 command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Enterprise\\VC\\Tools\\MSVC\\14.25.28610\\bin\\HostX86\\x64\\cl.exe' failed with exit status 2
> 23 rpy2/_rinterface_cffi_build.py:188: UserWarning: cannot parse "void __cffi_extern_python_start; void _capsule_finalizer(SEXP); void __cffi_extern_python_stop;"
> 24  .cdef source string. :500:96: Illegal character '\r'
> 25    warnings.warn(str(e))

But if you are not worried, why would I be ?

**Comment by lgautier** [comment]
It would be surprising if the `\r` error is specific to R-4.0.

Otherwise the error you have suggests a mismatch between the R library loaded at runtime and the one you'd expect to be loaded.

**Comment by HanOostdijk** [comment]
Thanks Laurent, this helps.

In the `path` I still referenced the old folder. Replacing this by `D:\tools\R\R-4.0.0\bin\x64\`  I no longer get the error messages. 

However two issues remain for me:

- as a result of `%load_ext rpy2.ipython` I get the warning in the notebook
> D:\tools\python\python38\Lib\site-packages\rpy2\robjects\packages.py : 365
> UserWarning: The symbol 'quartz' is not in this R namespace/package
>     warnings.warn(
-  the output of R 'sessions' does not appear in the notebook but in the command window where I started the notebook. As a newbie I would expect the result to be returned in the  notebook?

> %%R
> sessionInfo()

Output in command window (not in notebook) :
> 
> R version 4.0.0 (2020-04-24)
> Platform: x86_64-w64-mingw32/x64 (64-bit)
> Running under: Windows 10 x64 (build 18363)
> 
> Matrix products: default
> 
> locale:
> [1] LC_COLLATE=English_United States.1252
> [2] LC_CTYPE=English_United States.1252
> [3] LC_MONETARY=English_United States.1252
> [4] LC_NUMERIC=C
> [5] LC_TIME=English_United States.1252
> 
> attached base packages:
> [1] tools     stats     graphics  grDevices utils     datasets  methods
> [8] base
> 
> loaded via a namespace (and not attached):
> [1] compiler_4.0.0

**Comment by lgautier** [comment]
Getting rpy2 to work again on Windows is both recent, and not yet on par with Linux or OS X. Check issue #682 to see if you can find hints about what might be the issue on your end.

**Comment by HanOostdijk** [comment]
I already followed [#682](https://github.com/rpy2/rpy2/issues/682) but got no clues about what to do about it. 
My conclusion: at the moment my configuration is not supported in a way that I would like it to be and I can't do anything about it. Game over.

**Comment by HanOostdijk** [comment]
I saw today that version 3.3.3 was available. So I uninstalled version 3.3.2 and installed version 3.3.3 .
From the issues I described in https://github.com/rpy2/rpy2/issues/695#issuecomment-623092199 the most important is solved: the output of 
```
%%R
sessionInfo()
```
appears now in the notebook. Now it makes sense for me to explore the Python - R interface.

The minor issue that there is a strange warning at initialisation time remains:
```
%load_ext rpy2.ipython
d:\python38\lib\site-packages\rpy2\robjects\packages.py:365: UserWarning: The symbol 'quartz' is not in this R namespace/package.
  warnings.warn(
```
Thanks for the good work
